spec/unit/shoulda/matchers/active_record/validate_uniqueness_of_matcher_spec.rb

```diff
@@ -283,6 +283,11 @@
         column_type: :string
     end
 
+    context 'when one of the scoped attributes is a boolean column' do
+      include_context 'it supports scoped attributes of a certain type',
+        column_type: :boolean
```

Comment:
I think the tests for these are going to be a little different, because there's a limited number of values for a boolean column.

If you have one existing record where the scope attribute is either true or false, then you can simply flip that scope attribute. If you have two existing records where one of the record's scope attribute is true and the other is false... then you can't flip the scope attribute.

Say you have a validation on an attribute against a boolean scope:

```
class Person < ActiveRecord::Base
  validates :name, scope: :snowflake
end
```

Here's what a test would look like if there's only one existing record:

```
describe Person do
  it "is valid with a unique name when snowflake is true" do
    create(:person, name: "Steve", snowflake: true)

    person = build(:person, name: "Steve", snowflake: true)
    person.valid?
    expect(person).errors[:name].not_to be_empty

    person = build(:person, name: "Joe", snowflake: true)
    person.valid?
    expect(person).errors[:name].to be_empty

    # here we can test against the snowflake attribute -
    # i.e. if name stays same but snowflake is different
    person = build(:person, name: "Steve", snowflake: false)
    person.valid?
    expect(person).errors[:name].to be_empty
  end
end
```

But now consider if there are two existing records. Here's what those tests might look like:

```
describe Person do
  it "is valid with a unique name when snowflake is true" do
    create(:person, name: "Steve", snowflake: true)
    create(:person, name: "Steve", snowflake: false)

    person = build(:person, name: "Steve", snowflake: true)
    person.valid?
    expect(person).errors[:name].not_to be_empty

    person = build(:person, name: "Joe", snowflake: true)
    person.valid?
    expect(person).errors[:name].to be_empty

    # we can't really test what happens if name stays same and snowflake is different
    # b/c all values for snowflake are filled - so let's just test what happens if name is
    # different when snowflake is false
    person = build(:person, name: "Steve", snowflake: false)
    person.valid?
    expect(person).errors[:name].not_to be_empty

    person = build(:person, name: "Joe", snowflake: false)
    person.valid?
    expect(person).errors[:name].not_to be_empty
  end
end
```

So now the question is... how do we modify the matcher to run these tests. I don't have a clear solution for this, but it seems like as soon as one of the scopes is a boolean attribute, it kind of messes with the existing logic...